### PR TITLE
test(NODE-4918): import all driver APIs from one registry file

### PR DESCRIPTION
**Pull request by nbbeeken**
### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .evergreen/install-dependencies.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ set -o errexit  # Exit the script with error if any of the commands fail
 NVM_WINDOWS_URL="https://github.com/coreybutler/nvm-windows/releases/download/1.1.9/nvm-noinstall.zip"
 NVM_URL="https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh"
 
-NODE_LTS_NAME=${NODE_LTS_NAME:-erbium}
+NODE_LTS_NAME=${NODE_LTS_NAME:-fermium}
```
Comment (nbbeeken) [inline .evergreen/install-dependencies.sh:7]: No auth tests were still running on node 12

File: src/cmap/connection.ts [hunk 2]
```diff
@@ -95,22 +95,6 @@ export interface CommandOptions extends BSONSerializeOptions {
   writeConcern?: WriteConcern;
 }
 
-/** @internal */
-export interface GetMoreOptions extends CommandOptions {
```
Comment (nbbeeken) [inline src/cmap/connection.ts:99]: This interface was duplicated, caused an issue when merging all the imports to one file, I've de-duped (all internal, no public API impact)

File: src/sdam/server.ts [hunk 3]
```diff
@@ -503,7 +504,7 @@ function makeOperationHandler(
       return callback(undefined, result);
     }
 
-    if (options?.noResponse === true) {
+    if (options != null && 'noResponse' in options && options.noResponse === true) {
```
Comment (nbbeeken) [inline src/sdam/server.ts:507]: The noResponse property is only on CommandOptions

File: test/integration/node-specific/bson-options/utf8_validation.test.ts [hunk 4]
```diff
@@ -14,115 +14,98 @@ const EXPECTED_VALIDATION_ENABLED_ARGUMENT = {
...
+
+        await collection.insertOne(
+          { name: 'John Doe' },
+          passOptionTo === 'operation' ? option : {}
+        );
+
+        expect(onMessageSpy).to.have.been.called;
+        const binMsg = onMessageSpy.lastCall.firstArg;
+        const result = binMsg.parseBsonSerializationOptions(option);
+        expect(result).to.deep.equal(EXPECTED_VALIDATION_DISABLED_ARGUMENT);
```
Comment (nbbeeken) [inline test/integration/node-specific/bson-options/utf8_validation.test.ts:55]: b/c of the way we export things we cannot spy on bson serialize anymore. So instead I check our helper works as expected.

File: test/mongodb.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function printExports() {
```
Comment (nbbeeken) [inline test/mongodb.ts:5]: This can reprint the exports seen below if we need it

**Review by durran — APPROVED** [review]
Test failures look unrelated.

**Pull request merged**